### PR TITLE
Update testsuite

### PR DESCRIPTION
**Pull request by sbc100**
The main change here is the addition of declared elem sections.

### CODE REVIEW

File: src/interp/binary-reader-interp.cc [hunk 1]
```diff
@@ -645,6 +649,17 @@ bool BinaryReaderInterp::OnError(const Error& error) {
   return true;
 }
 
+wabt::Result BinaryReaderInterp::EndModule() {
```
Comment (binji) [inline src/interp/binary-reader-interp.cc]: could be `EndElemSection`, but this is fine too.

File: src/wast-parser.cc [hunk 2]
```diff
@@ -646,23 +659,31 @@ Result WastParser::ParseVarList(VarVector* out_var_list) {
 
 bool WastParser::ParseElemExprOpt(ElemExpr* out_elem_expr) {
   Location loc = GetLocation();
-  if (MatchLpar(TokenType::RefNull)) {
+  if (MatchLpar(TokenType::Item)) {
+    if (!ParseElemExprOpt(out_elem_expr))
```
Comment (binji) [inline src/wast-parser.cc]: This seems weird; we probably don't want this to be recursive, right?

File: src/wast-parser.cc [hunk 3]
```diff
@@ -355,6 +355,19 @@ void ResolveFuncTypes(Module* module) {
   }
 }
 
+void MarkDeclaredFunctions(Module* module) {
```
Comment (binji) [inline src/wast-parser.cc]: This logic should be in the validator, not the parser/binary-reader

File: src/interp/binary-reader-interp.cc (hunk 1 above)
Comment (sbc100) [reply src/interp/binary-reader-interp.cc]: Hmm, I tried that but it doesn't for if there are is no elem section at all.   Should we change the semantics so end for sections are called even if there is none?  I guess not.  I'll add a comment at least.

File: src/wast-parser.cc (hunk 3 above)
Comment (sbc100) [reply src/wast-parser.cc]: One problem with doing that is the validator generally doesn't mutate the IR (it takes a const Module).

I did consider doing this internally withing the validator using a side data structure rather then embedding this `is_declared` field in the IR itself.   Do you think that would be better approach?  I'll give that a go and see what you think..

File: src/interp/binary-reader-interp.cc (hunk 1 above)
Comment (binji) [reply src/interp/binary-reader-interp.cc]: Oh, good point. Yeah, it shouldn't call if the section is omitted.

File: src/wast-parser.cc (hunk 3 above)
Comment (binji) [reply src/wast-parser.cc]: Yeah, I'd prefer it that way I think. Storing it in the IR seems odd to me.

**Comment by sbc100** [comment]
Should be good to go now.

**Pull request merged**